### DISCUSSION TITLE
recreating pandoc [@citation] and @citation style with typst (ideally using the citation with brackets)

### DISCUSSION THREAD

**Discussion by danieltomasz** [typst/typst#4833] [Q&A] [answered]
I am coming from writing my thesis articles in markdown in pandoc where `[@jones2024]` was rendered in  author-date style as `(Jones et al, 2024)` and `jones2024` was rendered as `Jones et al. (2024)`;
https://pandoc.org/chunkedhtml-demo/8.20-citation-syntax.html

Currently Typst is supporting `@jones2024` that is rendered as `(Jones et al, 2024)`
I would like to still  use bracketed citation   `[@jones2024]` in Typst in similar way I am using  it via pandoc due to interoperability with other tools I am using to add citation; is there a possibility the change the behavior of Typst (via plugin or extension) to to make it more  similar to pandoc markdown?

**Comment by EpicEricEE** [comment, marked as answer]
You can set the [`cite.form` parameter](https://typst.app/docs/reference/model/cite/#parameters-form). To render the citation as `(Jones et al, 2024)`, you can set it to `"normal"` (or don't set it at all as it's the default) and to render it as `Jones et al. (2024)`, set it to `"prose"`. You can't use the `@citation` syntax though when you want to specify the form, but have to explicitly write `#cite(<citation>, form: ...)` (or create a custom function to simplify it).

To change the default behavior, you can apply the set rule `#set cite(form: "prose")`, then you have to specify it each time you want the normal behavior back, e.g.

```typ
#set cite(form: "prose")

This citation @jones2024 shows as Jones et al. (2024),
but this #cite(<jones2024>, form: "normal") shows as (Jones et al, 2024).
```

The bracketed syntax `[@jones2024]` is not really possible without resorting to some ugly methods (show rules on sequences).

**Reply by danieltomasz** [reply]
Thanks, this is useful!

**Reply by danieltomasz** [reply]
hi @EpicEricEE  maybe you might be able to help - would be possible have simple shortcut for rule prose?
for example  to define `~@jones2024` as style prose (with tilde), while leaving `@jones2024` as style normal by default?

**Reply by EpicEricEE** [reply]
No, at least not simple as it's basically the exact same thing as with brackets. The cite element (`@jones2024`) doesn't know what comes before or after it.
